### PR TITLE
[DM-28462] Improve EFD client docs

### PR DESCRIPTION
**Pull request by afausti**

### CODE REVIEW

**Review by SimonKrughoff — APPROVED** [review]
This looks great.  I added one commit to add a docstring to the `list_efd_names` class method since not having it was making that method not appear as prominently in the API docs

**Comment by afausti** [comment]
I can look at the build failures tomorrow.

File: lsst_efd_client/efd_utils.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def resample(df1, df2, interp_type='time'):
         That is the length of the resulting `pandas.DataFrame` is the
         sum of the lengths of the inputs.
     """
-    df = df1.append(df2, sort=False)  # Sort in this context does not sort the data
+    df = pandas.concat([df1, df2], axis=1)  # Sort in this context does not sort the data
```
Comment (afausti) [inline lsst_efd_client/efd_utils.py:76]: Does this do the trick?

**Review by afausti — COMMENTED** [review]
Nice, glad you figured it out. I have run the tests locally too! good job!

File: lsst_efd_client/efd_utils.py (hunk 1 above)
Comment (SimonKrughoff) [reply lsst_efd_client/efd_utils.py:76]: Yes.  That adds a column for ever entry in either dataframe with None filled out for the fields that don't exist.  Then the interpolate will fill the None values with the appropriate value given the interpolation type

**Pull request merged**